src/main/java/org/csanchez/jenkins/plugins/kubernetes/PodUtils.java

```diff
@@ -99,12 +101,16 @@ public static void cancelQueueItemFor(Pod pod, String reason) {
         for (Queue.Item item : q.getItems()) {
             Queue.Task task = item.task;
             if (runUrl.equals(task.getUrl())) {
-                LOGGER.log(Level.FINE, "Cancelling queue item: \"{0}\"\n{1}", new Object[] {
-                    task.getDisplayName(), !StringUtils.isBlank(reason) ? "due to " + reason : ""
-                });
-                q.cancel(item);
-                cancelled = true;
-                break;
+                String whyOnQueue = item.getWhy();
+                if(whyOnQueue.contains(metadata.getName())) {
```

Comment:
This seems fragile. We can probably match based on the label request instead?

Reply:
I tried printing a couple of "assignedLabels" for items on the queue ( return value of Queue.Item.getAssignedLabel() )
![image](https://github.com/jenkinsci/kubernetes-plugin/assets/1141454/d78f3469-75b9-4305-a8fa-5be5631bff14)

some of these assigned labels match the exactly the pod label "jenkins/label" 
but some other, have that `-` at the beginning that I don't really where it comes from 

![image](https://github.com/jenkinsci/kubernetes-plugin/assets/1141454/d139fa22-724d-4f63-b7fb-99d30b3d77f1)

![image](https://github.com/jenkinsci/kubernetes-plugin/assets/1141454/bc02a23e-17be-4dfa-b144-9123354d53ef)

Reply:
![image](https://github.com/jenkinsci/kubernetes-plugin/assets/1141454/cbf7d238-628b-4dcc-abb2-1d39af56260c)
but the assignedLabel value and the why String have the same value 

and the issue where jenkins/label does not match the assignedLabel String is in the case when the pod Name is using the fallback DEFAULT_AGENT_PREFIX ( because the given label does not match the regex set here https://github.com/jenkinsci/kubernetes-plugin/blob/40d4fb21bd8046082ecd88fc5cf860f0db8e54ee/src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java#L313C14-L313C23 )

Reply:
so, the question I have is why the queue.item assignedLabel not the pod name but the value of the label in jenkins/label instead..